app/code/Magento/Indexer/Test/Unit/Model/ProcessManagerTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Indexer\Test\Unit\Model;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Indexer\Model\ProcessManager;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class covers process manager execution test logic
+ */
+class ProcessManagerTest extends TestCase
+{
+    /**
+     * @dataProvider closureFunctionsProvider
+     * @param array $userFunctions
+     * @param int $threadsCount
+     * @return void
+     */
+    public function testFailureInChildProcessHandleMultiThread(array $userFunctions, int $threadsCount): void
+    {
+        $connectionMock = $this->createMock(ResourceConnection::class);
+        $processManager = new ProcessManager(
+            $connectionMock,
+            null,
+            $threadsCount
+        );
+
+        try {
+            $processManager->execute($userFunctions);
+            $this->fail('Exception was not handled');
+        } catch (\RuntimeException $exception) {
+            $this->assertEquals('Fail in child process', $exception->getMessage());
+        }
+    }
+
+    /**
+     * Closure functions data provider for multi thread execution
+     *
+     * @return array
+     * @SuppressWarnings(PHPMD.ExitExpression)
+     */
+    public function closureFunctionsProvider(): array
+    {
+        return [
+            'more_threads_than_functions' => [
```

Comment:
Could you also add coverage for cases when it's mess threads than functions and when it's equal?

Reply:
thanks for looking into it, will do

Reply:
added variations